common/fallback/src/lib.rs

```diff
@@ -32,6 +42,67 @@ impl<T> Fallback<T> {
         Err(FallbackError::AllErrored(errors))
     }
 
+    /// Run the given function for all servers concurrently and stop as soon as one of them
+    /// successfully returns. If no server returns successfully wait for all servers and return all
+    /// errors.
+    pub async fn first_success_concurrent<'a, F, O, E, R>(
+        &'a self,
+        func: F,
+    ) -> Result<O, FallbackError<(usize, E)>>
+    where
+        F: Fn(&'a T) -> R,
+        R: Future<Output = Result<O, E>>,
+    {
+        let mut errors = Vec::with_capacity(self.servers.len());
+        let mut open_futures: Vec<_> = self
+            .servers
+            .iter()
+            .map(|s| Box::pin(func(s).fuse()))
```

Comment:
I think you are right and we can omit the `fuse` here. I added it because it is a requirement for the `select!` macro (see https://docs.rs/futures/0.3.8/futures/macro.select.html) but didn't check if it is also a requirement for `select_all`. Will remove it, good catch!